arch/risc-v/src/bl602/bl602_pwm_lowerhalf.c

```diff
@@ -0,0 +1,473 @@
+/****************************************************************************
+ * arch/risc-v/src/bl602/bl602_pwm_lowerhalf.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <arch/board/board.h>
+
+#include "riscv_arch.h"
+#include "riscv_internal.h"
+
+#include "bl602_gpio.h"
+#include "bl602_pwm_lowerhalf.h"
+#include "hardware/bl602_pwm.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define BL_PWM_FREQ_DEFAULT       (1000)
+#define BL_PWM_STOP_TIMEOUT_COUNT (160*1000)
+
+#define BL_PWM_XTAL_CLK           (40000000)
+#define BL_PWM_BUS_BCLK           (80000000)
+
+#ifdef CONFIG_PWM_NCHANNELS
+#  if (CONFIG_PWM_NCHANNELS > 5)
+#    error "The maximum number of BL602 PWM channels is 5!"
+#  endif
+#endif
+
+#ifdef CONFIG_BL602_PWM0_BUS_BCLK_SRC
```

Comment:
Is `CONFIG_BL602_PWM0_BUS_BCLK_SRC` configured anywhere? I expected to see it in a Kconfig, but can't find it.

Reply:
@liangzhanggb please take a look.